File: backend/app/messaging/models/message.py
from sqlalchemy import Column, Integer, String, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from app.db.base_class import Base
import datetime

class Message(Base):
    __tablename__ = "messages"

    id = Column(Integer, primary_key=True, index=True)
    content = Column(String, index=True)
    timestamp = Column(DateTime, default=datetime.datetime.utcnow)
    sender_id = Column(Integer, ForeignKey("users.id"))
    recipient_id = Column(Integer, ForeignKey("users.id"))
    conversation_id = Column(Integer, ForeignKey("conversations.id"))

    sender = relationship("User", foreign_keys=[sender_id])
    recipient = relationship("User", foreign_keys=[recipient_id])
    conversation = relationship("Conversation", back_populates="messages")

File: server/components/communication/models/Message.js
/**
 * Communication Component - Message Model
 * Collection: 'communication/conversations/messages'
 * Fields: id, senderId, conversationId, content, timestamp, readBy
 */
module.exports = {
  collection: 'messages',
  fields: ['senderId', 'conversationId', 'content', 'timestamp']
};
